osbs/core.py

```diff
@@ -373,6 +373,15 @@ def get_build_config_by_labels_filtered(self, label_selectors, filter_key, filte
             raise OsbsException(
                 "More than one build config found for labels: %r" %
                 (label_selectors, ))
+        elif filter_value:
+            raise OsbsException(
+                "No match for {0} found for labels: %r".format(filter_value) %
+                (label_selectors, ))
```

Comment:
Should we also raise an exception if there is more than one match after applying the filter? That's what the docstring says we do.

I feel like the function should be structured a bit more like:
- get the build configs that match the label selectors
- apply the filter, if there is a filter to apply
- raise an exception if we don't have a single build config
- return it

Reply:
Restructured as requested.